components/core/src/GlobalMySQLMetadataDB.hpp

```diff
@@ -66,7 +66,7 @@ class GlobalMySQLMetadataDB : public GlobalMetadataDB {
                               uint64_t size) override;
     void update_metadata_for_files (const std::string& archive_id, const std::vector<streaming_archive::writer::File*>& files) override;
 
-    GlobalMetadataDB::ArchiveIterator* get_archive_iterator () override;
+    GlobalMetadataDB::ArchiveIterator* get_archive_iterator (epochtime_t begin_ts, epochtime_t end_ts) override;
```

Comment:
Instead of modifying the existing iterator, could we create a new one for filtering by timestamps? E.g., `get_archive_iterator_for_time_range`

Reply:
I've added a new iterator `get_archive_iterator_for_time_window`